### PR TITLE
feat(telephony.portability.document): add upload document for portability

### PR DESCRIPTION
**Pull request by Steffy29**
Ref: UXCT-34

### CODE REVIEW

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  const interceptor = {
+    response(response) {
+      cache.remove(response.config.url);
+      queryCache.removeAll();
+      return response.resource;
+    },
+  };
+
+  const docResource = $resource('/telephony/:billingAccount/portability/:id/document', {
```
Comment (antleblanc) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: Shouldn't be:

```suggestion
  const docResource = $resource('/telephony/:billingAccount/portability/:id/document/:documentId', {
```

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      method: 'GET',
+      isArray: true,
+    },
+    create: {
+      method: 'POST',
+      interceptor,
+    },
+    getDocument: {
+      method: 'GET',
+      url: '/telephony/:billingAccount/portability/:id/document/:documentId',
```
Comment (antleblanc) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: `url` property not needed (related to my previous comment).

```suggestion
```

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      isArray: false,
+    },
+    updateDocument: {
+      method: 'PUT',
+      url: '/telephony/:billingAccount/portability/:id/document/:documentId',
```
Comment (antleblanc) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]:
```suggestion
```

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      interceptor,
+    },
+    deleteDocument: {
+      method: 'DELETE',
+      url: '/telephony/:billingAccount/portability/:id/document/:documentId',
```
Comment (antleblanc) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]:
```suggestion
```

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      interceptor,
+    },
+    cors: {
+      method: 'POST',
+      url: '/telephony/:billingAccount/portability/:id/document/cors',
```
Comment (antleblanc) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: This URL doesn't exist on the API.

Shouldn't be done via the `create` action?

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      return response.resource;
+    },
+  };
+
+  const docResource = $resource('/telephony/:billingAccount/portability/:id/document', {
+    billingAccount: '@billingAccount',
+    id: '@id',
+    documentId: '@documentId',
+  }, {
+    query: {
```
Comment (marie-j) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: This is the default behaviour for `$resource.query` if you don't override it there is no need to declare it

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      origin: $window.location.origin,
+    }).$promise.then(() => $http.put(resp.putUrl, file, {
+      serviceType: 'storage',
+      headers: {
+        'Content-type': 'multipart/form-data',
+      },
+    }).then(() => docResource.get({
+      id: resp.id,
+    }).$promise)));
+  };
```
Comment (marie-j) [inline src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: I'm not too sure this should be handled here. The call chain is part of the product logic and the aim here is only to reflect API

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js (hunk 5 above)
Comment (Steffy29) [reply src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: yes I removed it

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js (hunk 7 above)
Comment (Steffy29) [reply src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: I took example on /me/document to upload file.

File: src/api/telephony/portability/document/telephony-portability-document.v6.service.js (hunk 6 above)
Comment (Steffy29) [reply src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: @marie-j Does $resource.query return an array by default?

Comment (antleblanc) [reply src/api/telephony/portability/document/telephony-portability-document.v6.service.js]: @Steffy29 Yes it is (https://docs.angularjs.org/api/ngResource/service/$resource#$resource-returns)

**Pull request merged**